src/Listener/PodeRequest.cs

```diff
@@ -421,41 +360,78 @@ protected virtual bool ValidateInput(byte[] bytes)
         /// </summary>
         public virtual void PartialDispose()
         {
-            if (BufferStream != default(MemoryStream))
+            try
             {
-                BufferStream.Dispose();
-                BufferStream = default;
-            }
+                if (BufferStream != default(MemoryStream))
+                {
+                    BufferStream.Dispose();
+                    BufferStream = default;
+                }
 
-            Buffer = default;
+                // Clear the contents of the Buffer array
+                if (Buffer != null)
+                {
+                    Array.Clear(Buffer, 0, Buffer.Length);
+                }
+
+            }
+            catch (Exception ex)
+            {
+                PodeHelpers.WriteException(ex, Context.Listener, PodeLoggingLevel.Error);
+            }
         }
 
         /// <summary>
-        /// Disposes of the request and its associated resources.
+        /// Dispose managed and unmanaged resources.
         /// </summary>
-        public virtual void Dispose()
+        /// <param name="disposing">Indicates if disposing is called manually or by garbage collection.</param>
+        protected virtual void Dispose(bool disposing)
```

Comment:
I don't see the need for having `disposing` and the deconstructor below?

If GC comes along, surely we want it to dispose of the Stream/Socket? And in the logic below, even if `disposing=false` we still set `IsDisposed=true` anyway 🤔 

It's also better for `IsDisposed=true` to be set further up, before we actually dispose of objects, so we're not disposing of objects and then finally setting it to true - which does open the possibility for `ObjectDisposedException` as `Receive` could use them

Reply:
You're absolutely right to question the need for both the destructor and the disposing = false logic. Since the class only deals with managed resources like streams and sockets, the destructor (~PodeRequest) isn't necessary. 
Removing it simplifies things and avoids the disposing = false scenario altogether.
As for setting IsDisposed = true earlier—great point! Doing that upfront helps prevent potential ObjectDisposedException during concurrent use by marking the object as disposed before starting cleanup.
😊

Reply:
regarding why to use `Dispose(bool disposing)`
https://learn.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
Aaah, never knew about the hidden `dispose(bool)`. If I'm reading it correctly then in, for example, `PodeHttpRequest` the `RawBody` and `_body` variables being nullied should be outside of the `if (disposing)` since they're unmanaged, while `BodyStream`/`Form`/etc should be within the `if` because they're managed? 🤔

Reply:
it's not a requirement. All resources are disposed at the end doesn't matter if best practice are followed or not